modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/state/ResourceOwnerOAuthContext.java

```diff
@@ -47,4 +52,26 @@
    */
   String getResourceOwnerId();
 
+  /**
+   * @return the state of the dance for the resource owner of this context.
+   */
+  DancerState getDancerState();
+
+  /**
+   * @param lockNamePrefix a prefix to uniquely identify the locks in the provided {@code lockFactory}.
+   * @param lockFactory the object to get the locks from. This will ensure that the same lock instance is used after
+   *        serialization/deserialization of this context.
+   * @return a lock that can be used to avoid concurrency problems trying to update oauth context.
+   */
+  Lock getRefreshUserOAuthContextLock(String lockNamePrefix, LockFactory lockFactory);
```

Comment:
add a description to the Javadoc. Also, why is the word `User` in this method name?